s4e-backend/src/main/java/pl/cyfronet/s4e/ex/ErrorHandlerHelper.java

```diff
@@ -52,6 +52,30 @@
         return map;
     }
 
+    public Map<String, Object> toResponseMap(QueryException e) {
+        BindingResult bindingResult = e.getBindingResult();
+        val map = new LinkedHashMap<String, Object>();
+
+        // populate general errors
+        if (bindingResult.hasGlobalErrors()) {
+            map.put("__general__", bindingResult.getGlobalErrors().stream()
+                    .map(objectError -> getMessage(objectError))
+                    .collect(Collectors.toUnmodifiableList()));
+        }
+
+        // populate field errors
+        for (FieldError fieldError : bindingResult.getFieldErrors()) {
+            map.putIfAbsent(fieldError.getField(), new ArrayList<>());
+            ((List) map.get(fieldError.getField())).add(
+                    messageSource.getMessage(fieldError.getCode(),
+                    fieldError.getArguments(),
+                    LocaleContextHolder.getLocale()));
+        }
```

Comment:
Could you move this if to function `populateFieldErrors`?

Reply:
I would keep it there, as it's not so complicated, and there are only two places it's used in, see "rule of three"

Reply:
The root of the idea is: put blocks of code in more contextual places - methods - (instead of adding comments like `populate general errors` above them). In this case, the comment wasn't caused by duplication. 

In general: `abstraction should go from top to down`, where detailed method name should describe what they're exactly doing inside. Those methods with comments are a perfect example of it.